### PR TITLE
chore: extract backend server util

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/backend.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        ...process.env,
+        ...this._envProvider(),
+      },
+    });
+
+    const client = this._clientFactory();
+
+    serverProcess.stderr?.on('data', () => {});
+    serverProcess.on('error', error => {
+      client?._onErrorEvent.fire(error);
```
Comment (dgozman) [inline src/backend.ts:53]:
```suggestion
      client._onErrorEvent.fire(error);
```

File: src/backend.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        const match = data.toString().match(/Listening on (.*)/);
+        if (!match)
+          return;
+        const wse = match[1];
+        await client._connect(wse);
+        fulfill(client);
+      });
+      serverProcess.on('exit', () => {
+        fulfill(null);
+        client?._onCloseEvent.fire();
```
Comment (dgozman) [inline src/backend.ts:67]:
```suggestion
        client._onCloseEvent.fire();
```

File: src/backend.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        error.stack = message.error.error?.stack;
+        pair.reject(error);
+      } else {
+        pair.fulfill(message.result);
+      }
+    };
+    await this.initialize();
+  }
+
+  async initialize() { }
```
Comment (dgozman) [inline src/backend.ts:123]: Perhaps we can do `abstract` here to force subclasses to implement it.

**Review by dgozman — APPROVED** [review]

File: src/backend.ts (hunk 3 above)
Comment (pavelfeldman) [reply src/backend.ts:123]: not everybody needs it.

**Pull request merged**